src/InternalKeyboardEvents.tsx

```diff
@@ -4,10 +4,23 @@ import useKBar from "./useKBar";
 
 type Timeout = ReturnType<typeof setTimeout>;
 
-export default function InternalKeyboardEvents() {
-  const { query, options, visualState, actions } = useKBar((state) => ({
+export default function InternalEvents() {
+  return (
+    <>
+      <ToggleHandler />
+      <ShortcutsHandler />
+      <FocusHandler />
+      <DocumentLock />
```

Comment:
Have some separation of concerns for each type of event, not fully set on this but seem quite nice & simple.

Reply:
Interesting pattern, if all of these `return null` and reason not to use hook-style?

Reply:
Can totally convert these to hooks style, will do. Do you know of a better pattern we can use to register these different event listeners?

Reply:
Not particularly, I'd _consider_ depending on another library for the keyboard shortcuts functionality though as you go forward – I feel like there are bunch of edge cases to consider that don't need to be dealt with in kbar.

Reply:
Going to convert these to hooks style for now and get this in as a first step – will further look into external libs for keyboard shortcuts, want to really get this right 💯